server/handlers/handlers.go

```diff
@@ -60,3 +60,37 @@ func PlayerScores(game *stycoon.Game, w http.ResponseWriter, req *http.Request)
 	}
 	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
 }
+
+func Commands(game *stycoon.Game, w http.ResponseWriter, req *http.Request) {
+	// read
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		log.Warn().Err(err).Msg("Error reading body")
+		http.Error(w, "can't read request body", http.StatusBadRequest)
+		return
+	}
+	// parse
+	commands, err := stycoon.ParseCommands(string(body))
+	if err != nil {
+		log.Warn().Err(err).Msg("Command parsing failed")
+		http.Error(w, "parsing failed", http.StatusBadRequest)
+		return
+	}
+	// process
+	errs := game.ProcessCommands(commands)
+	for _, _ = range errs {
+		b, err := json.Marshal(errs)
+		if err != nil {
+			log.Warn().Err(err).Msg("Json marshall failed")
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+		_, err = w.Write(b)
```

Comment:
you cannot do it like this. Prepare the response, serialize it to json and send.

Not sure what this is going to do (probably mash the jsons after each other) but it is not what you want.

Also if you want to return some response you have to define it in the API

Reply:
The loop runs only once. Changed it to if statement. The response is defined in the API, look for CommandError (it's a simple map,  that's why there was no structure generated)